rslib/src/scheduler/new.rs

```diff
@@ -119,12 +120,16 @@ impl Collection {
             let cards = col.storage.all_searched_cards_in_search_order()?;
             for mut card in cards {
                 let original = card.clone();
-                card.schedule_as_new(position);
+                if let Some(original_position) = card.original_position {
+                    card.schedule_as_new(original_position);
+                } else {
+                    card.schedule_as_new(position);
+                    position += 1;
+                }
```

Comment:
While you said the main motivation is shared decks, I guess this could also be convenient and won't hurt?

Reply:
It may depend on the user's preference - I suspect some users like to be able to throws cards to the back of the queue, but I imagine others would want the original position restored instead. May be worth giving users the choice in the future?

Reply:
Letting the users decide sounds good, I've reverted it for now.